qiskit/qpy/binary_io/circuits.py

```diff
@@ -594,12 +597,13 @@ def _write_instruction(file_obj, instruction, custom_operations, index_map, use_
         # ucr*_dg gates can have different numbers of parameters,
         # the uuid is appended to avoid storing a single definition
         # in circuits with multiple ucr*_dg gates.
-        if instruction.operation.name in ["ucrx_dg", "ucry_dg", "ucrz_dg"]:
-            gate_class_name += "_" + str(uuid.uuid4())
-
-        if gate_class_name not in custom_operations:
-            custom_operations[gate_class_name] = instruction.operation
-            custom_operations_list.append(gate_class_name)
+        if instruction.operation.name in {"ucrx_dg", "ucry_dg", "ucrz_dg"}:
+            gate_class_name = f"{gate_class_name}_{uuid.uuid4()}"
+        elif version >= 11:
+            # Assign a uuid to each instance of a custom operation
+            gate_class_name = f"{gate_class_name}_{uuid.uuid4().hex}"
```

Comment:
Maybe the order of the if/elif could be reverted? Not that it would change anything in the output, because the uuid is not outputted anymore, but it feels nicer to know that all the uuids are in the same format if version>=11.

Reply:
Oh hah, I did it this way on purpose to preserve the exact behavior because I didn't view the ucr* gates as being custom gates and didn't want to overreach in this PR and change more in the format revision than was necessary.  But you're right we can just subsume the custom logic for the ucr* gates in version 11 and have it all be consistent. I'll reverse the logic here

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/11646/commits/c88a9a2784eb511e571eae00bd1e81541fcd4ecd